lib/image-stream/utils.js

```diff
@@ -16,47 +16,30 @@
 
 'use strict';
 
-const _ = require('lodash');
 const Bluebird = require('bluebird');
-const fs = Bluebird.promisifyAll(require('fs'));
-const archiveType = require('archive-type');
+const mime = require('mime-types');
 
 /**
  * @summary Get archive mime type
  * @function
  * @public
  *
- * @param {String} file - file path
+ * @param {String} filename - file path
  * @fulfil {String} - mime type
  * @returns {Promise}
  *
  * @example
- * utils.getArchiveMimeType('path/to/raspberrypi.img.gz').then((mimeType) => {
- *   console.log(mimeType);
- * });
+ * const mimeType = utils.getArchiveMimeType('path/to/raspberrypi.img.gz');
  */
-exports.getArchiveMimeType = (file) => {
-
-  // `archive-type` only needs the first 261 bytes
-  // See https://github.com/kevva/archive-type
-  const ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH = 261;
-
-  return Bluebird.using(fs.openAsync(file, 'r').disposer((fileDescriptor) => {
-    return fs.closeAsync(fileDescriptor);
-  }), (fileDescriptor) => {
-    const BUFFER_START = 0;
-    const chunk = new Buffer(ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH);
+exports.getArchiveMimeType = (filename) => {
+  // NOTE: For some awkward reason `mime-db` (used by `mime-types`)
+  // does not contain a file extension for `.gz`, so we need to test for it here
+  // TODO: Remove this once https://github.com/jshttp/mime-db/pull/68 is merged
+  if (/\.gz$/i.test(filename)) {
+    return 'application/gzip';
+  }
 
-    return fs.readAsync(
-      fileDescriptor,
-      chunk,
-      BUFFER_START,
-      ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH,
-      null
-    ).then(() => {
-      return _.get(archiveType(chunk), [ 'mime' ], 'application/octet-stream');
-    });
-  });
+  return mime.lookup(filename) || 'application/octet-stream';
```

Comment:
> Yeah, I was wondering that too! Eventually figured out (by looking through git history) that it's due to https://github.com/sindresorhus/file-type/blob/master/index.js#L175

Oh, I missed that one! Well, that's not going to fail if there are fewer than 4100 bytes, and archive-type would hit that case as well, because it only filters for archive extensions after calling `fileType()`.

> to prevent it mis-detecting a DMG as a BZIP2.

Yeah, exactly – and DMGs are also not the only format being an elusive bugger with magic number detection – so this way around we should also avoid those automagically, should they arise.

> But as I mentioned, do we really need to pull in all of mime-types just to determine whether a file ends with .dmg or not??

I'd recommend it, rather than hand-rolling it, especially since it has pretty good reusability and we (hopefully) won't have to think about this again when adding new formats.